src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckCodeFixProvider.cs

```diff
@@ -102,5 +117,64 @@ public MyCodeAction(Func<CancellationToken, Task<Document>> createChangedDocumen
             {
             }
         }
+
+        private class FirstStatementTracker
+        {
+            private readonly Collection<StatementSyntax> _removed;
+            public ICollection<StatementSyntax> FirstStatements { get; }
```

Comment:
you do containment and removal, but no need for anything else.  consider making into a HashSet to better represent this and guarantee good `O(...)` behavior